content/automate/ManageIQ/Transformation/Ansible.class/__methods__/launchplaybookasaservice.rb

```diff
@@ -27,7 +27,10 @@ def main
             return if service_template.nil?
             target_host = target_host(task, transformation_hook)
             return if target_host.blank?
-            service_dialog_options = { :hosts => target_host.ipaddresses.first }
+            service_dialog_options = {
+              :credential => service_template.options[:config_info][:provision][:credential_id],
```

Comment:
The `[:config_info]` hash of `service_template` has an accessor and we should use that.

Use either
`service_template.config_info.dig(:provision, :credential_id)`  (If you expect any of the hash levels to be nil)
or
`service_template.config_info[:provision][:credential_id]`

service_template.options[:config_info][:provision][:credential_id]`

Reply:
Done. Thanks.